### ISSUE TITLE
Use type hinting with mypy

### ISSUE THREAD

**Issue by cwmeijer** [dianna-ai/dianna#85]
It is getting a bit convoluted what the input types of specific functions are. We now have options: numpy array, xarray, string, list of strings. I think it would help both development and users to specify at certain places what types we expect.

**Comment by egpbos** [comment]
Indeed, probably all the places where we currently feel that we have to specify it in the docstring (input arguments like `matrix (np.array): blabla`) we should have it as a type hint instead.

**Comment by elboyran** [comment]
Seems important as we are adding soon more modalities.

**Comment by stefsmeets** [comment]
We discussed this today in the retrospective. We agreed this is a good idea to add gradually.

My suggestion would be to just enable mypy and then see how it goes.

@geek-yang @cwmeijer
